--- .local/share/nvim/plugged/vim-quickui/autoload/quickui/menu.vim	2025-03-02 16:15:10.146651010 +0300
+++ .local/share/nvim/plugged/vim-quickui/autoload/quickui/menu.vim	2025-03-02 16:21:56.214650855 +0300
@@ -724,7 +724,7 @@
 	let s:cmenu.hotkey = s:cmenu.inst.hotkey
 	let s:cmenu.state = 1
 	let s:cmenu.context = -1
-	let s:cmenu.next = 0
+	let s:cmenu.next = get(a:opts, 'next', 0)
 	let keymap = quickui#utils#keymap()
 	let s:cmenu.keymap = keymap
 	let s:cmenu.script = ''
@@ -775,9 +775,7 @@
 		endif
 	endwhile
 	call nvim_win_close(winid, 0)
-	redraw
-	echo ""
-	redraw
+	Showtab
 	let s:namespace[name].index = s:cmenu.index
 	if s:cmenu.script != ''
 		let script = s:cmenu.script
