--- .local/share/nvim/site/pack/pckr/opt/playground/plugin/nvim-treesitter-playground.lua	2025-02-12 14:55:06.361884596 +0300
+++ .local/share/nvim/site/pack/pckr/opt/playground/plugin/nvim-treesitter-playground.lua	2025-02-12 14:56:29.221884564 +0300
@@ -1,3 +1,4 @@
+if plugin_installed('nvim-treesitter-playground') then
 -- setup playground module
 require("nvim-treesitter-playground").init()
 local api = vim.api
@@ -26,3 +27,4 @@
 api.nvim_create_user_command("TSHighlightCapturesUnderCursor", function()
   require("nvim-treesitter-playground.hl-info").show_hl_captures()
 end, {})
+end
