--- .local/share/nvim/site/pack/pckr/opt/playground/lua/nvim-treesitter-playground.lua	2025-02-12 14:52:53.825884646 +0300
+++ .local/share/nvim/site/pack/pckr/opt/playground/lua/nvim-treesitter-playground.lua	2025-02-12 14:51:34.533884676 +0300
@@ -1,3 +1,4 @@
+if plugin_installed('nvim-treesitter') then
 local parsers = require "nvim-treesitter.parsers"
 local M = {}
 
@@ -32,3 +33,4 @@
 end
 
 return M
+end
