--- .local/share/nvim/site/pack/pckr/opt/telescope.nvim/lua/telescope/pickers.lua	2025-02-12 14:33:54.101885081 +0300
+++ .local/share/nvim/site/pack/pckr/opt/telescope.nvim/lua/telescope/pickers.lua	2025-02-12 14:34:20.629885071 +0300
@@ -2,6 +2,10 @@
 
 local a = vim.api
 
+if not plugin_installed('plenary') then
+	goto exit
+end
+do
 local async = require "plenary.async"
 local await_schedule = async.util.scheduler
 local channel = require("plenary.async.control").channel
@@ -1727,3 +1731,5 @@
 pickers._Picker = Picker
 
 return pickers
+end
+::exit::
